backend/src/app.ts

```diff
@@ -943,4 +946,147 @@ app.post('/api/add-contact-question', authenticate, async (req, res) => {
   }
 });
 
+// 1. Fix the camelCase interface properties
+interface TravelTimes {
+  artsQuadWalking: number;
+  artsQuadDriving: number;
+  duffieldWalking: number;
+  duffieldDriving: number;
+  ctbWalking: number;
+  ctbDriving: number;
+}
+
+const {GOOGLE_MAPS_API_KEY} = process.env;
+const LANDMARKS = {
+  arts_quad: 'Arts Quad, Cornell University, Ithaca, NY 14853',
+  duffield: 'Duffield Hall, Cornell University, Ithaca, NY 14853',
+  ctb: 'College Town Bagels, College Ave, Ithaca, NY 14850',
+};
+
+interface TravelTimeInput {
+  origin: string; // Can be either address or "latitude,longitude"
+}
+
+/**
+ * Travel Times Calculator - Calculates walking and driving times from a given origin to Cornell landmarks.
+ *
+ * @remarks
+ * Uses Google Maps Distance Matrix API to calculate travel times to three landmarks: Arts Quad,
+ * Duffield Hall, and College Town Bagels. Returns both walking and driving durations in minutes.
+ * Origin can be either an address or coordinates in "latitude,longitude" format.
+ *
+ * @route POST /api/travel-times
+ *
+ * @input {string} req.body.origin - Starting point, either as address or "latitude,longitude"
+ *
+ * @status
+ * - 200: Successfully retrieved travel times
+ * - 400: Missing or invalid origin
+ * - 500: Server error or Google Maps API failure
+ */
+app.post('/api/travel-times', async (req, res) => {
+  try {
+    const { origin } = req.body as TravelTimeInput;
+
+    if (!origin) {
+      return res.status(400).json({ error: 'Origin is required' });
+    }
+
+    // Create destinations array
+    const destinations = Object.values(LANDMARKS);
+
+    // Make request to Google Distance Matrix API for walking times
+    const walkingResponse = await axios.get(
+      `https://maps.googleapis.com/maps/api/distancematrix/json?origins=${encodeURIComponent(
+        origin
+      )}&destinations=${destinations
+        .map((dest) => encodeURIComponent(dest))
+        .join('|')}&mode=walking&key=${GOOGLE_MAPS_API_KEY}`
+    );
+
+    // Make request to Google Distance Matrix API for driving times
+    const drivingResponse = await axios.get(
+      `https://maps.googleapis.com/maps/api/distancematrix/json?origins=${encodeURIComponent(
+        origin
+      )}&destinations=${destinations
+        .map((dest) => encodeURIComponent(dest))
+        .join('|')}&mode=driving&key=${GOOGLE_MAPS_API_KEY}`
+    );
+
+    // Fix the any types with a proper interface
+    interface DistanceMatrixElement {
+      duration: {
+        value: number;
+      };
+    }
+
+    const walkingTimes = walkingResponse.data.rows[0].elements.map(
+      (element: DistanceMatrixElement) => element.duration.value / 60
+    );
+    const drivingTimes = drivingResponse.data.rows[0].elements.map(
+      (element: DistanceMatrixElement) => element.duration.value / 60
+    );
```

Comment:
Similarly, even though the code looks great, a small thing that could be done to improve it would be to make a function that will help reduce code duplication.